be/src/exec/hdfs_scanner.cpp

```diff
@@ -375,27 +385,38 @@ void HdfsScannerContext::update_materialized_columns(const std::unordered_set<st
     }
 
     materialized_columns.swap(updated_columns);
+    return Status::OK();
 }
 
-void HdfsScannerContext::update_not_existed_columns_of_chunk(ChunkPtr* chunk, size_t row_count) {
-    if (not_existed_slots.empty() || row_count <= 0) return;
-
+void HdfsScannerContext::append_or_update_not_existed_columns_to_chunk(ChunkPtr* chunk, size_t row_count) {
+    if (not_existed_slots.empty() || row_count < 0) return;
     ChunkPtr& ck = (*chunk);
+    ck->set_num_rows(row_count);
     for (auto* slot_desc : not_existed_slots) {
-        ck->get_column_by_slot_id(slot_desc->id())->append_default(row_count);
+        auto col = ColumnHelper::create_column(slot_desc->type(), slot_desc->is_nullable());
+        col->append_default(row_count);
+        if (ck->is_slot_exist(slot_desc->id())) {
+            ck->update_column(std::move(col), slot_desc->id());
+        } else {
+            ck->append_column(std::move(col), slot_desc->id());
+        }
     }
 }
 
-void HdfsScannerContext::append_not_existed_columns_to_chunk(ChunkPtr* chunk, size_t row_count) {
-    if (not_existed_slots.size() == 0) return;
-
+void HdfsScannerContext::append_or_update_count_column_to_chunk(ChunkPtr* chunk, size_t row_count) {
```

Comment:
could it happen when use count optimize, there is other not existed solots?

Reply:
if there is ___count___column, this will be the only one existed column.